File: Aula_Ricardo/Programação_orientada_a_objetos/go/Métodos.go
package main

import "fmt"

type Carro struct {
	Marca      string
	Modelo     string
	Ano        int
	Velocidade int
}

func (c Carro) ExibirDetalhes() {
	fmt.Printf("Marca: %s, Modelo: %s, Ano: %d\n", c.Marca, c.Modelo, c.Ano)
}

func (c *Carro) Acelerar(incremento int) {
	c.Velocidade += incremento
}

func (c *Carro) Frear(decremento int) {
	if c.Velocidade-decremento < 0 {
		c.Velocidade = 0
	} else {
		c.Velocidade -= decremento
	}
}

func (c Carro) ExibirVelocidade() {
	fmt.Printf("A velocidade atual é de %d km/h\n", c.Velocidade)
}

func main() {
	carro1 := Carro{Marca: "Toyota", Modelo: "Land Cruiser", Ano: 2010}
	carro2 := Carro{Marca: "Honda", Modelo: "Civic", Ano: 2018}

	carro1.ExibirDetalhes()
	carro2.ExibirDetalhes()

	carro1.Acelerar(70)
	carro1.ExibirVelocidade()

	carro1.Frear(25)
	carro1.ExibirVelocidade()

	carro2.Acelerar(120)
	carro2.ExibirVelocidade()
}

File: Aula_Ricardo/Programação_orientada_a_objetos/java/Métodos.java
public class Carro {
    private String marca;
    private String modelo;
    private int ano;
    private int velocidade;

    public Carro(String marca, String modelo, int ano) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.velocidade = 0;
    }

    public void exibirDetalhes() {
        System.out.println("Marca: " + marca + ", Modelo: " + modelo + ", Ano: " + ano);
    }

    public void acelerar(int incremento) {
        velocidade += incremento;
    }

    public void frear(int decremento) {
        velocidade = Math.max(0, velocidade - decremento);
    }

    public void exibirVelocidade() {
        System.out.println("A velocidade atual é de " + velocidade + " km/h");
    }

    public static void main(String[] args) {
        Carro carro1 = new Carro("Toyota", "Land cruiser", 2010);
        Carro carro2 = new Carro("Honda", "Civic", 2018);

        carro1.exibirDetalhes();
        carro2.exibirDetalhes();

        carro1.acelerar(70);
        carro1.exibirVelocidade();
        carro1.frear(25);
        carro1.exibirVelocidade();

        carro2.acelerar(120);
        carro2.exibirVelocidade();
    }
}
